connection_pool.go

```diff
@@ -49,21 +51,59 @@ func NewConnectionPool(addresses []HostAddress, conf PoolConfig, log Logger) (*C
 		addresses: convAddress,
 		hostIndex: 0,
 	}
-	if err = newPool.initPool(); err != nil {
+
+	// Init pool with non-SSL socket
+	if err = newPool.initPool(false, nil); err != nil {
 		return nil, err
 	}
 	newPool.startCleaner()
 	return newPool, nil
 }
 
-// initPool innitializes the connection pool
-func (pool *ConnectionPool) initPool() error {
+// NewConnectionPool constructs a new SSL connection pool using the given addresses and configs
+func NewSslConnectionPool(addresses []HostAddress, conf PoolConfig, sslConfig *tls.Config, log Logger) (*ConnectionPool, error) {
```

Comment:
Maybe upper the `Ssl` is better ? 
`NewSSLConnection`

Reply:
I saw Ssl in used in some other package like https://www.eclipse.org/jetty/javadoc/jetty-10/org/eclipse/jetty/io/ssl/SslClientConnectionFactory.html.

Reply:
ok